### PR TITLE
Test code - GpuMatND & Mat interoperability

### PR DESCRIPTION
**Pull request by nglee**
**Main PR**: https://github.com/opencv/opencv/pull/19259

```
opencv=
force_builders=Custom
buildworker:Custom=linux-4
build_image:Custom=ubuntu-cuda:18.04
```

### CODE REVIEW

**Comment by alalek** [comment]
BTW, Please avoid using of `#` in commit messages (GitHub tries to create wrong cross-links in this case)

File: modules/cudev/test/test_nd.cu [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        GpuMatND gmat;
+
+        {
+            // simple upload, download test for GpuMatND
+
+            gmat.upload(gold);
+            gmat.download(dst);
+
+            EXPECT_TRUE(std::equal(gold.begin(), gold.end(), dst.begin()));
+        }
```
Comment (asmorkalov) [inline modules/cudev/test/test_nd.cu:48]: It'll be great if you split the blocks like this on dedicated parameterized tests with GTest. It simplifies issue triaging and reduces side effects if one block affects another during execution. The test you used as base was implemented in prehistoric period and we try to get rid of large 'doTest' things that does everything.

Comment (nglee) [reply modules/cudev/test/test_nd.cu:48]: Currently, it uses type-parameterized tests. Should I change it to a value parameterized form?

Comment (asmorkalov) [reply modules/cudev/test/test_nd.cu:48]: Up to you.

Comment (nglee) [reply modules/cudev/test/test_nd.cu]: I have fixed it.

File: modules/cudev/test/test_nd.cu [hunk 2]
```diff
@@ -0,0 +1,247 @@
+#include "test_precomp.hpp"
```
Comment (alalek) [inline modules/cudev/test/test_nd.cu:1]: Please use short license header: https://github.com/opencv/opencv/wiki/Coding_Style_Guide#file-structure

```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.
```

File: modules/cudev/test/test_nd.cu [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+using namespace cv;
+using namespace cv::cuda;
+using namespace cvtest;
```
Comment (alalek) [inline modules/cudev/test/test_nd.cu:5]: Please use `namespace opencv_test { namespace {` like others tests do.

File: modules/cudev/test/test_nd.cu [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+    static MatType RandomMat(const SizeArray& size)
+    {
+        const auto dims = static_cast<int>(size.size());
+
+        MatType ret(dims, size.data());
+
+        for (ElemType& elem : ret)
+            for (int i = 0; i < cn; ++i)
+                elem[i] = static_cast<CnType>(std::rand());
```
Comment (alalek) [inline modules/cudev/test/test_nd.cu:24]: `cv::randu()` ?

Comment (nglee) [reply modules/cudev/test/test_nd.cu]: I have fixed test codes but builds failed because of a merge conflict.

**Review by alalek — APPROVED** [review]

**Pull request merged**